File: src-legacy/rewrite-1/yellow/ui/dialogs/AchievementListDialog.java
package yellow.ui.dialogs;

import arc.scene.style.*;
import arc.scene.ui.*;
import arc.util.*;
import mindustry.gen.*;
import mindustry.graphics.*;
import mindustry.ui.dialogs.*;

public class AchievementListDialog extends BaseDialog{

    private static final String lockName = "[gray]???[]";
    private static final String lockDesc = "[gray]< locked >[]";

    public AchievementListDialog(){
        super("@yellow.achievements");
        addCloseButton();
    }

    @Override
    public Dialog show(){
        cont.clear();

        cont.pane(paneTable -> {
            YellowAchievement.instances.each(achievement -> {
                paneTable.table(Tex.pane, achievementInfo -> {
                    achievementInfo.table(Tex.pane, iconspace -> {
                        iconspace.image(() -> ((TextureRegionDrawable) (achievement.isUnlocked() ? achievement.unlockIcon : achievement.lockIcon)).getRegion()).scaling(Scaling.bounded);
                    }).left().size(140f);

                    achievementInfo.table(infospace -> {
                        infospace.center();
                        infospace.defaults().center();
                        infospace.labelWrap(() -> achievement.isUnlocked() ? achievement.displayName : lockName).labelAlign(Align.center).padBottom(4f).row();
                        infospace.image().color(Pal.accent).height(5f).growX().row();

                        infospace.pane(description -> {
                            description.center();
                            description.defaults().center();
                            description.labelWrap(
                                    () -> achievement.isUnlocked() ? achievement.description : (lockDesc + "\n" + achievement.hint)
                            ).labelAlign(Align.center);
                        }).grow();
                    }).grow().padLeft(5f);
                }).growX().height(170f).row();
            });
        }).grow();

        return super.show();
    }
}

File: src/yellow/ui/buttons/dialogs/AchievementListDialog.kt
package yellow.ui.buttons.dialogs

import arc.scene.ui.Dialog
import arc.util.Scaling
import com.github.mnemotechnician.mkui.extensions.dsl.*
import mindustry.gen.Tex
import mindustry.graphics.Pal
import mindustry.ui.dialogs.BaseDialog
import yellow.game.YAchievement

@Suppress("LeakingThis")
open class AchievementListDialog: BaseDialog("Achievements") {
    init {
        addCloseButton()
    }

    override fun show(): Dialog{
        cont.clear()

        cont.scrollPane {
            YAchievement.instances.each {ach ->
                addTable {
                    background = Tex.pane

                    addTable(Tex.pane) {
                        addImage(if(ach.isUnlocked()) ach.icon else ach.iconLocked, Scaling.bounded)
                    }.left().size(140f)

                    addTable {
                        addLabel(if(ach.isUnlocked()) ach.displayName else "[darkgray]???[]").growX().padBottom(4f).row()
                        image().color(Pal.accent).height(5f).growX().row()
                        scrollPane {
                            addLabel(if(ach.isUnlocked()) ach.description else "[darkgray]< locked >[]\n${ach.hint}", wrap = true).grow()
                        }.grow()
                    }.grow().padLeft(5f)
                }.growX().height(170f).row()
            }
        }.grow()

        return super.show()
    }
}